### DISCUSSION TITLE
Picking audio track in a multi-audio files

### DISCUSSION THREAD

**Discussion by Bearmancer** [openai/whisper#2083] [Q&A]
I want to transcribe the director's commentary in a particular mkv file instead of the default audio. How do I choose a different audio track in Whisper?

**Comment by glangford** [comment]
You can extract the relevant audio track using ffmpeg. See this gist for examples.

* https://gist.github.com/cdeath/1f8fdd96ead3829465897e413188190a

Other tools are also available. whisper itself does not support track selection.

**Comment by Bearmancer** [comment]
The help page was confusing so I asked ChatGPT for help and made a pwsh script to extract the second audio track whenever applicable as a 16-bit FLAC file in a directory for MKV files.

For those interested, here is the script:

`function ExtractCommentaryAudio() {
    $videoFiles = Get-ChildItem -Path . -Recurse -Filter "*.mkv"

    foreach ($file in $videoFiles) {
        $audioTrackCount = (ffmpeg -i $file.FullName -hide_banner 2>&1 | Select-String "Stream #.*: Audio").Count

        if ($audioTrackCount -gt 1) {
            $outputFile = Join-Path $PWD.Path ("$($file.BaseName) - Audio Commentary.flac")
            ffmpeg -i $file.FullName -map 0:a:1 -sample_fmt s16 -acodec flac $outputFile
        
        }
    }

}`
